### PR TITLE
Move Generator ops to c10

### PR DESCRIPTION
**Pull request by smessmer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26434 Move Generator ops to c10**

Differential Revision: [D17465434](https://our.internmc.facebook.com/intern/diff/D17465434/)

### CODE REVIEW

File: aten/src/ATen/core/jit_type.h [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ template <typename T>
 struct getTypePtr_ final {
   static TypePtr call() {
     if (!isCustomClassRegistered<T>()) {
-      throw c10::Error("Type could not be converted to any of the known types.", "");
+      throw c10::Error(std::string() + "Type " + typeid(T).name() + " could not be converted to any of the known types.", "");
```
Comment (ezyang) [inline aten/src/ATen/core/jit_type.h]: Oy! RTTI!!

**Review by ezyang — CHANGES_REQUESTED** [review]
RTTI

File: aten/src/ATen/core/jit_type.h (hunk 1 above)
Comment (smessmer) [reply aten/src/ATen/core/jit_type.h]: good catch. Seems I forgot to remove my debug code...

I'm actually hoping to enable this after the gcc5 switch using compile time type identification,  having a better error message here would be really nice, but that's separate from this PR.

**Comment by smessmer** [comment]
@ezyang can you take another look?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**